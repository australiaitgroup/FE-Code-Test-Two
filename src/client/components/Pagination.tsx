import * as React from 'react';
import MediaQuery from 'react-responsive';
import { animateScroll } from 'react-scroll';
import { Container, Pagination } from 'semantic-ui-react';

import { IPaginationProps } from './interfaces/pagination.interface';

const PaginationComponent: React.FunctionComponent<IPaginationProps> = ({
    handlePageChange,
    nbPages,
    page,
}) => nbPages
    ? (
        <Container textAlign="center">
            <MediaQuery minWidth={768}>
                {(matches) => {
                    const boundaryRange = matches ? 1 : 0;
                    const siblingRange = matches ? 1 : 0;

                    return (
                        <Pagination
                            activePage={page}
                            boundaryRange={boundaryRange}
                            onPageChange={(event, data) => {
                                animateScroll.scrollToTop({ duration: 400 });
                                handlePageChange(data.activePage as number);
                            }}
                            siblingRange={siblingRange}
                            size="mini"
                            totalPages={nbPages}
                        />
                    );
                }}
            </MediaQuery>
        </Container>
    )
    : null;

export default PaginationComponent;
