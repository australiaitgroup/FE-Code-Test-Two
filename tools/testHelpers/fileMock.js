// Used by Jest: refer to https://jestjs.io/docs/en/webpack.html

const path = require('path');

module.exports = {
    process(src, filename) {
        return `module.exports = ${JSON.stringify(path.basename(filename))};`;
    },
};
