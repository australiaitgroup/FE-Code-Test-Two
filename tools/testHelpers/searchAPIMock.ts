import * as nock from 'nock';

export default nock('https://hn.algolia.com/api/v1');
