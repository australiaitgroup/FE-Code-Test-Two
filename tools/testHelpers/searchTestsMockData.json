{
  "searchResult": {
    "hits": [
      {
        "created_at": "2015-06-03T19:45:33.000Z",
        "title": null,
        "url": null,
        "author": "pingec",
        "points": null,
        "story_text": null,
        "comment_text": "test",
        "num_comments": null,
        "story_id": 9654758,
        "story_title": "How Well Can You Hear Audio Quality?",
        "story_url": "http://www.npr.org/sections",
        "parent_id": 9654758,
        "created_at_i": 1433360733,
        "objectID": 32223433,
        "_highlightResult": {
          "author": {
            "matchLevel": "none",
            "matchedWords": [],
            "value": "pingec"
          },
          "comment_text": {
            "fullyHighlighted": false,
            "matchLevel": "full",
            "matchedWords": ["testttd"],
            "value": "Similar test: <a href='http://test.tidalhifi.com'"
          },
          "story_title": {
            "matchLevel": "none",
            "matchedWords": [],
            "value": "How Well Can You Hear Audio Quality?"
          },
          "story_url": {
            "matchLevel": "none",
            "matchedWords": [],
            "value": "http://www.npr.org/sections/therecord/2015/06/02/"
          }
        },
        "_tags": ["comment", "author_pingec", "story_9654758"]
      }
    ],
    "nbHits": 1,
    "page": 0,
    "nbPages": 1,
    "hitsPerPage": 20,
    "processingTimeMS": 10,
    "exhaustiveNbHits": true,
    "query": "test",
    "params": "advancedSyntax=true&analytics=false&hitsPerPage=20&query=testttd&tags=%28story%2Ccomment%29"
  }
}
