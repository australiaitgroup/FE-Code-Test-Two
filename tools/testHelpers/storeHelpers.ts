import { Store } from 'redux';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import reducers from '../../src/client/reducers';
import { IState } from '../../src/types/stateAndAction';

export function getInitialState(): IState {
    return reducers(undefined, { type: null });
}

export function createMockStore(initialState: IState): Store {
    const middlewares = [thunk];
    const mockStore = configureStore(middlewares);
    return mockStore(initialState);
}

export function deepClone(toBeCloned: any): any {
    return JSON.parse(JSON.stringify(toBeCloned));
}
