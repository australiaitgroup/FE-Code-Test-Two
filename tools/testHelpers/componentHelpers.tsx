import * as React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import * as renderer from 'react-test-renderer';

import { createMockStore } from './storeHelpers';

export function renderComponent(children, storeData) {
    const store = createMockStore(storeData);

    return renderer.create(
        <Provider store={store}>
            <BrowserRouter>
                {children}
            </BrowserRouter>
        </Provider>);
}
