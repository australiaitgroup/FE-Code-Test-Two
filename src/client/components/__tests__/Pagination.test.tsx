import * as React from 'react';

import Pagination from '../Pagination';
import { renderComponent } from '../../../../tools/testHelpers/componentHelpers';

it('Pagination renders correctly', () => {
    const props = {
        handlePageChange: () => Promise.resolve(),
        nbPages: 50,
        page: 3,
    };

    const component = renderComponent(<Pagination {...props} />, {});
    const result = component.toJSON();
    expect(result).toMatchSnapshot();
});
