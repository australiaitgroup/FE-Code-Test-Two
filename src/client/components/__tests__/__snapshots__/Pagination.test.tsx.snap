// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination renders correctly 1`] = `
<div
  className="ui center aligned container"
>
  <div
    aria-label="Pagination Navigation"
    className="ui mini pagination menu"
    role="navigation"
  >
    <a
      aria-current={false}
      aria-label="First item"
      className="item"
      onClick={[Function]}
      onKeyDown={[Function]}
      tabIndex={0}
      type="firstItem"
      value={1}
    >
      «
    </a>
    <a
      aria-current={false}
      aria-label="Previous item"
      className="item"
      onClick={[Function]}
      onKeyDown={[Function]}
      tabIndex={0}
      type="prevItem"
      value={2}
    >
      ⟨
    </a>
    <a
      aria-current={false}
      className="disabled item"
      onClick={[Function]}
      onKeyDown={[Function]}
      tabIndex={-1}
      type="ellipsisItem"
      value={2}
    >
      ...
    </a>
    <a
      aria-current={true}
      className="active item"
      onClick={[Function]}
      onKeyDown={[Function]}
      tabIndex={0}
      type="pageItem"
      value={3}
    >
      3
    </a>
    <a
      aria-current={false}
      className="disabled item"
      onClick={[Function]}
      onKeyDown={[Function]}
      tabIndex={-1}
      type="ellipsisItem"
      value={4}
    >
      ...
    </a>
    <a
      aria-current={false}
      aria-label="Next item"
      className="item"
      onClick={[Function]}
      onKeyDown={[Function]}
      tabIndex={0}
      type="nextItem"
      value={4}
    >
      ⟩
    </a>
    <a
      aria-current={false}
      aria-label="Last item"
      className="item"
      onClick={[Function]}
      onKeyDown={[Function]}
      tabIndex={0}
      type="lastItem"
      value={50}
    >
      »
    </a>
  </div>
</div>
`;
